In [1]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [2]:
%cd /content/drive/MyDrive/Dibimbing/Day 8 - Data Manipulation

/content/drive/MyDrive/Dibimbing/Day 8 - Data Manipulation


In [3]:
%ls

'Data Manipulation - Module_Student_v2.ipynb'
'Day 8 - Data Manipulation.pdf'
 Day_8_Hands_on_Data_Manipulation.ipynb
 telco_customer_churn.csv


In [4]:
import pandas as pd

# Task 1 - Load Data to DataFrame

In [5]:
df = pd.read_csv('telco_customer_churn.csv')
df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7043 rows x 21 columns]

In [6]:
df.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   
3         Yes          No              No        One year               No   
4          No          No              No  Month-to-month              Yes   

               PaymentMethod MonthlyCharges  TotalCharges Churn  
0           Electronic check          29.85         29.85    No  
1               Mailed check          56.95        1889.5    No  
2               Mailed check          53.85        108.15   Yes  
3  Bank transfer (automatic)          42.30       1840.75    No  
4           Electronic check          70.70        151.65   Yes  

[5 rows x 21 columns]

In [7]:
df.tail()

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[5 rows x 21 columns]

# Task 2 - DataFrame Information

Show total data, mean, min, max, quartil1, median, quartil3 from **tenure** column

In [8]:
df.describe()

SeniorCitizen       tenure  MonthlyCharges
count    7043.000000  7043.000000     7043.000000
mean        0.162147    32.371149       64.761692
std         0.368612    24.559481       30.090047
min         0.000000     0.000000       18.250000
25%         0.000000     9.000000       35.500000
50%         0.000000    29.000000       70.350000
75%         0.000000    55.000000       89.850000
max         1.000000    72.000000      118.750000

In [9]:
## for tenure column
df[['tenure']].describe()

tenure
count  7043.000000
mean     32.371149
std      24.559481
min       0.000000
25%       9.000000
50%      29.000000
75%      55.000000
max      72.000000

Total data (count) = 7043

Mean = 32.371149

Min = 0

Max = 72

Quartil 1 (25%) = 9

Median (50%) = 29

Quartil 3 (75%) = 55


# Task 3 - Comparison

Show all data that have tenure more than 55 or
which have MonthlyCharges more than 70

In [10]:
df[(df['tenure'] > 55) | (df['MonthlyCharges'] > 70)]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
6     1452-KIOVK    Male              0      No        Yes      22   
8     7892-POOKP  Female              0     Yes         No      28   
9     6388-TABGU    Male              0      No        Yes      62   
...          ...     ...            ...     ...        ...     ...   
7037  2569-WGERO  Female              0      No         No      72   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService MultipleLines InternetService       OnlineSecurity  ...  \
4             Yes            No     Fiber optic                   No  ...   
5             Yes           Yes     Fiber optic                   No  ...   
6             Yes           Yes     Fiber optic                   No  ...   
8             Yes           Yes     Fiber optic                   No  ...   
9             Yes            No             DSL                  Yes  ...   
...           ...           ...             ...                  ...  ...   
7037          Yes            No              No  No internet service  ...   
7038          Yes           Yes             DSL                  Yes  ...   
7039          Yes           Yes     Fiber optic                   No  ...   
7041          Yes           Yes     Fiber optic                   No  ...   
7042          Yes            No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
4                      No                   No                   No   
5                     Yes                   No                  Yes   
6                      No                   No                  Yes   
8                     Yes                  Yes                  Yes   
9                      No                   No                   No   
...                   ...                  ...                  ...   
7037  No internet service  No internet service  No internet service   
7038                  Yes                  Yes                  Yes   
7039                  Yes                   No                  Yes   
7041                   No                   No                   No   
7042                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
4                      No  Month-to-month              Yes   
5                     Yes  Month-to-month              Yes   
6                      No  Month-to-month              Yes   
8                     Yes  Month-to-month              Yes   
9                      No        One year               No   
...                   ...             ...              ...   
7037  No internet service        Two year              Yes   
7038                  Yes        One year              Yes   
7039                  Yes        One year              Yes   
7041                   No  Month-to-month              Yes   
7042                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
4              Electronic check          70.70        151.65   Yes  
5              Electronic check          99.65         820.5   Yes  
6       Credit card (automatic)          89.10        1949.4    No  
8              Electronic check         104.80       3046.05   Yes  
9     Bank transfer (automatic)          56.15       3487.95    No  
...                         ...            ...           ...   ...  
7037  Bank transfer (automatic)          21.15        1419.4    No  
7038               Mailed check          84.80        1990.5    No  
7039    Credit card (

There are 4223 datas that have tenure more than 55 or
which have MonthlyCharges more than 70

# Task 4 - Grouping

## a. gender

In [12]:
df_gender = df[['customerID','gender']].groupby(by='gender').count()
df_gender

customerID
gender            
Female        3488
Male          3555

In [14]:
## sort results from the largest number
df_gender.sort_values('customerID', ascending = False)

customerID
gender            
Male          3555
Female        3488

## b. InternetService

In [15]:
df_InternetService = df[['customerID','InternetService']].groupby(by='InternetService').count()
df_InternetService

customerID
InternetService            
DSL                    2421
Fiber optic            3096
No                     1526

In [16]:
## sort results from the largest number
df_InternetService.sort_values('customerID', ascending = False)

customerID
InternetService            
Fiber optic            3096
DSL                    2421
No                     1526

## c. PaymentMethod

In [18]:
df_PaymentMethod = df[['customerID','PaymentMethod']].groupby(by='PaymentMethod').count()
df_PaymentMethod

customerID
PaymentMethod                        
Bank transfer (automatic)        1544
Credit card (automatic)          1522
Electronic check                 2365
Mailed check                     1612

In [19]:
## sort results from the largest number
df_PaymentMethod.sort_values('customerID', ascending = False)

customerID
PaymentMethod                        
Electronic check                 2365
Mailed check                     1612
Bank transfer (automatic)        1544
Credit card (automatic)          1522

# Task 5 - Binning

In [20]:
df['ChargesType'] = pd.cut(df['MonthlyCharges'],
                            bins=[0,35.5,70.5,89.9,99999],
                            labels=['Low','Medium','High','Very High'])

In [21]:
df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0             No          No              No  Month-to-month              Yes   
1             No          No              No        One year               No   
2             No          No              No  Month-to-month              Yes   
3            Yes          No              No        One year               No   
4             No          No              No  Month-to-month              Yes   
...          ...         ...             ...             ...              ...   
7038         Yes         Yes             Yes        One year              Yes   
7039          No         Yes             Yes        One year              Yes   
7040          No          No              No  Month-to-month              Yes   
7041          No          No              No  Month-to-month              Yes   
7042         Yes         Yes             Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges TotalCharges  Churn ChargesType  
0              Electronic check          29.85        29.85     No         Low  
1                  Mailed check          56.95       1889.5     No      Medium  
2                  Mailed check          53.85       108.15    Yes      Medium  
3     Bank transfer (automatic)          42.30      1840.75     No      Medium  
4              Electronic check          70.70       151.65    Yes        High  
...                         ...            ...          ...    ...         ...  
7038               Mailed check          84.80       1990.5     No        High  
7039    Credit card (automatic)         103.20       7362.9     No   Very High  
7040           Electronic check          29.60       346.45     No         Low  
7041               Mailed check          74.40        306.6    Yes        High  
7042  Bank transfer (automatic)         105.65       6844.5     No   Very High  

[7043 rows x 22 columns]